### PR TITLE
docs: Add a pull request template

### PR DESCRIPTION
**Pull request by swickersh**
- Pull request template to help contributors avoid common omissions when submitting pull requests.

### CODE REVIEW

**Comment by swickersh** [comment]
@mmalina @johnbieren  No worries if you guys can't use this. I thought it might help avoid situations where something is merged with an open dependency.

**Comment by swickersh** [comment]
By the way, this  would automatically display that text template in all new pull requests text box.

File: pull_request_template.md
Comment (mmalina) [inline pull_request_template.md]: What is a `depends on` label? Is that something that Github knows about and will block on? I did a quick google and it seems it's something that people requested, but probably not something that Github supports? So I wonder how it would really help - other than perhaps the person merging the PR may notice.

So I think if it doesn't really do anything, opening a thread somewhere appropriate is probably more reliable - open threads do block merging of PRs.

**Review by johnbieren — COMMENTED** [review]
I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference :sweat_smile:

**Comment by johnbieren** [comment]
> I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference 😅

If we do decide to go with this approach, we should probably also mention modifying the version string and adding a `Changes in` entry in the README.md

**Comment by swickersh** [comment]
> I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference 😅

Yea I think the main advantage of this is that it's more in the user's face when they open a PR. (see screenshots)
I think adding bullets to the CONTRIBUTING.md wouldn't do much good if there's not more of a push for people to actually read it in the first place. 
image

image

File: pull_request_template.md
Comment (swickersh) [reply pull_request_template.md]: I thought we could make any custom label we wanted. Maybe `donotmerge` would be a better label.  People have created workflows to prevent merging [conditionally based on labels. ](https://stackoverflow.com/questions/71502652/prevent-merging-of-branch-conditionally-based-on-label)

But really I think it's just more of a visual reminder for the maintainers of this repo.  `draft` would accomplish the same thing and actually blocks, so that's probably just as good.

**Comment by swickersh** [comment]
> > I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference 😅
> 
> If we do decide to go with this approach, we should probably also mention modifying the version string and adding a `Changes in` entry in the README.md

I agree. I think title prefix should be added somewhere to.  I only see it in this regex check: https://github.com/konflux-ci/release-service-catalog/blob/13fe5adecd8409d707c46c7bd7a31d22052a15b7/.github/gitlint/contrib_format_conventional_commits.py#L18

File: pull_request_template.md
Comment (mmalina) [reply pull_request_template.md]: Yeah, I think using Draft is usually preferred, but I get that sometimes people want to get their PR reviewed even if they know they need to update the image once the utils PR is merged. And when you create a PR as a Draft, reviewers are not automatically added.

I wasn't aware you could accomplish a blocking label using github workflows, thanks for sharing. I've seen people add "do not merge" label and I do agree that it helps at least visually to make it clear it's not meant to be merged.

**Comment by mmalina** [comment]
> > > I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference 😅
> > 
> > 
> > If we do decide to go with this approach, we should probably also mention modifying the version string and adding a `Changes in` entry in the README.md
> 
> I agree. I think title prefix should be added somewhere to. I only see it in this regex check:
> 
> https://github.com/konflux-ci/release-service-catalog/blob/13fe5adecd8409d707c46c7bd7a31d22052a15b7/.github/gitlint/contrib_format_conventional_commits.py#L18

I agree that the prefixes are not well documented now - I found myself looking at the python file in the past, when I wasn't sure which prefixes are allowed. But similar to what Johnny said, I'd say we just need to add a note about this to the CONTRIBUTING guide.

File: pull_request_template.md
Comment (johnbieren) [reply pull_request_template.md]: You can add reviewers to a draft PR though. I do this often

**Comment by johnbieren** [comment]
> > I don't disagree with this, but I wonder if it is not better to just put bullets 1 and 2 in the CONTRIBUTING doc. 3 and 4 are kind of duplicated by CONTRIBUTING now (3 was recently removed as a requirement and from CONTRIBUTING). So if people are missing the final bullet, they aren't reading CONTRIBUTING.md, so I wonder if adding a new doc saying go read CONTRIBUTING.md will make much difference 😅
> 
> Yea I think the main advantage of this is that it's more in the user's face when they open a PR. (see screenshots) I think adding bullets to the CONTRIBUTING.md wouldn't do much good if there's not more of a push for people to actually read it in the first place. image
> 
> image

Oh that's fancy, I didn't know about this. Okay, I am cool with adding one of these files then

File: pull_request_template.md
Comment (mmalina) [reply pull_request_template.md]: Totally. But for a newcomer / external contributor, it might not be obvious whom to add.

Comment (johnbieren) [reply pull_request_template.md]: release-service-maintainers. The same as is auto added if a PR is opened. If they want review on their draft, they can add the whole team or the specific person they want

Comment (mmalina) [reply pull_request_template.md]: I know this of course. But an external contributor may not.

Comment (johnbieren) [reply pull_request_template.md]: What I am advocating for is to add it to this document. "If you want reviews on your draft PR, you can add reviewers or add the handle if you are unsure who to tag"

Comment (swickersh) [reply pull_request_template.md]: Added changes to address some of this

**Review by johnbieren — DISMISSED** [review]

File: CONTRIBUTING.md
Comment (mmalina) [inline CONTRIBUTING.md]: Can we have just one empty line here?

File: CONTRIBUTING.md
Comment (mmalina) [inline CONTRIBUTING.md]:
```suggestion
The title prefix should be one of (`chore`|`docs`|`feat`|`fix`|`refactor`|`revert`|`style`|`test`) followed by a colon (`:`) and lowercase title. Optionally, you can include a Jira key.

Examples:

- fix(KFLUXSPRT-794): pass content-gateway token
- feat: add rpms-signature-scan task

Title prefixes:
```

Comment (swickersh) [reply CONTRIBUTING.md]: Added, thanks

File: CONTRIBUTING.md
Comment (swickersh) [reply CONTRIBUTING.md]: Done. Thanks

**Review by johnbieren — APPROVED** [review]
Needs rebase

**Review by johnbieren — APPROVED** [review]

**Comment by johnbieren** [comment]
/ok-to-test
e2e does nothing here but if it has a successful e2e run when we merge it so I don't have to override, I won't complain :slightly_smiling_face:

**Review by mmalina — APPROVED** [review]

**Pull request merged**
